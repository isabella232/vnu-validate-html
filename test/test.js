'use strict';

const path = require('path');
const vnuValidateHtml = require('..');

const fixture = name => path.join(__dirname, 'fixtures', name + '.html');

describe('validate', () => {
  test('basically works', () => {
    return vnuValidateHtml.validate(fixture('one')).then(report => {
      expect(report).toMatchSnapshot();
    });
  });

  test('strips colors', () => {
    return vnuValidateHtml
      .validate(fixture('one'), {
        stripColors: true
      })
      .then(report => {
        expect(report).toMatchSnapshot();
      });
  });
});
