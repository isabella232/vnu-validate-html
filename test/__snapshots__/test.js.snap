// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate basically works 1`] = `
"
[4mtest/fixtures/one.html[24m [90m[[39m[35m12:3[39m[90m-[39m[35m12:6[39m[90m][39m
No “p” element in scope but a “p” end tag seen.
[31m⚆_⚆[39m Are you illegally nesting a block-level element inside a “p” element?
  [90mv</div>
    [39m[33m</p>[39m[90m
  </bod[39m
"
`;

exports[`validate strips colors 1`] = `
"
test/fixtures/one.html [12:3-12:6]
No “p” element in scope but a “p” end tag seen.
⚆_⚆ Are you illegally nesting a block-level element inside a “p” element?
  v</div>
    </p>
  </bod
"
`;
